docs/docs/developers/tutorials/codealong/first_fees.md

```diff
@@ -0,0 +1,354 @@
+
+The equivalent using aztec.js:
+
+```javascript
+import { FPCContract } from "@aztec/noir-contracts.js/FPC";
+  //...
+  const fpc = await FPCContract.deploy(wallets[0], bananaCoin.address, wallets[0].getAddress()).send().deployed();
+  // ... (setup token authorisation, and fpc balance here)
+  const publicFee = new PublicFeePaymentMethod(fpc.address, newWallet)
+
+```
+
+:::note  Payment: Fee Paying Contract (public/private)
+Options for payment via fpc that can be used in multiple commands:
+CLI: `--payment method=fpc-public,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS` (`fpc-private` for private)
+.js: `{ fee: { paymentMethod: new PublicFeePaymentMethod(fpc.address, newWallet) }` (`PrivateFeePaymentMethod` for private)
+:::
+
+:::info Private fee payment
+Note: using a private fpc method is the only way for transactions to be paid for privately.
+
+Public and private refer to how the fpc will claim its tokens from the sender to then (publicly) spend fee juice. The visibility of the function being called is unchanged.
+:::
+
+## Summary of fee payment options
+
+The two key ways of paying for transactions: fee juice from an account or via an FPC. Both of which are contracts on the aztec network.
+
+### Fee juice from an account (default)
+
+- from the sender of a transaction (default)
+  - most common if making transactions from a funded account
+  - difficult to deploy without funds already
+- specifying a different fee-payer account (new account creation/deployment only)
+  - using a funded account to deploy another account, or rapid testing on the sandbox
+  - need to already have a funded account
+- claiming fee juice from already-bridged L1 asset and immediately using
+  - great for bootstrapping an account on Aztec
+  - need to already have aztec fee asset on an L1 account
+
+### Fee juice from a fee paying contract
+
+- via public/private payment
+  - can pay in other assets, and not hold fee juice
+  - can privately spend alternative assets, to make private transactions
+- sponsored payment
+  - enables new users to bootstrap their first account
+
+### Example payment options
+
+```bash
+aztec-wallet <command> --payment method=fee_juice # default fee juice from sender of transaction
+aztec-wallet <create/deploy command> --payment method=fee_juice,feePayer=$FEE_PAYER_ADDRESS # fee juice from another account (create/deploy txs only)
+aztec-wallet <command> --payment method=fee_juice,claim # claim bridged fee juice and pay
+aztec-wallet <command> --payment method=fpc-sponsored,fpc=$SPONSORED_FPC_ADDRESS # sponsored fpc
+aztec-wallet <command> --payment method=fpc-public,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS # asset for fee juice via fpc public
+aztec-wallet <command> --payment method=fpc-private,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS # asset for fee juice via fpc private
+
+```
+
+```javascript
+`command({ fee: { paymentMethod: new FeeJuicePaymentMethod(<sender address>) }})` // default fee juice from sender of transaction
```

Comment:
I don't think you need the ` since its already wrapped in a code block. and it might be easier to read these code snippets with some line breaks and putting the comments in between the lines.

Reply:
Updated.